utils/docker/Dockerfile.centos.7

```diff
@@ -9,16 +9,72 @@
 FROM centos:centos7
 LABEL maintainer="daos@daos.groups.io"
 
+# Need to be able to use a local repository for packages
+ARG REPO_URL=""
+ARG REPO_DISTRO=""
+ARG REPO_DAOS=""
+
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release
+# package must be installed and the GPG key it provides must be
+# installed.
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/* ;                         \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      yum -y install --disablerepo=extras                       \
+             dnf dnf-pluins-core epel-release-7-11;             \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7;         \
+      yum -y install dnf dnf-plugins-core;                      \
+      dnf config-manager --assumeyes --quiet                    \
+                         --disable base extras epel updates;    \
+      yum clean all;                                            \
```

Comment:
I would prefer the clean before the disable, even though one is dnf and the other yum I do know they share some configuration.

Reply:
Neither am I, but this is the cautions approach.

Reply:
> Not sure if yum and dnf are sharing the same cache.

They are not.